packages/kbn-eslint-plugin-disable/src/rules/no_protected_eslint_disable.ts

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { REPO_ROOT } from '@kbn/utils';
+import { relative } from 'path';
+import Eslint from 'eslint';
+import { getReportLocFromComment, getRulesBlockFromEslintDisableComment } from '../helpers';
+
+export const PROTECTED_DISABLE_MSG_ID = 'no-protected-eslint-disable';
+const messages = {
+  [PROTECTED_DISABLE_MSG_ID]:
+    "The rule '{{ disabledRuleName }}' is protected and disabling it is not allowed. Please remove it from the statement.",
+};
+
+const meta: Eslint.Rule.RuleMetaData = {
+  type: 'problem',
+  fixable: 'code',
+  docs: {
+    description: 'Prevents the disabling of protected rules within eslint-disable* comments.',
+  },
+  messages,
+  schema: {
+    type: 'array',
+    uniqueItems: true,
+    items: {
+      type: ['string', 'array'],
+      items: {
+        properties: {
+          allowed: {
+            type: 'array',
+            uniqueItems: true,
+            items: {
+              type: 'string',
+            },
+          },
+        },
+      },
+    },
+  },
+};
+
+const getProtectedRulesFromOptions = function (ruleOptions: any[]) {
+  const protectedRules: { [key: string]: string | string[] } = {};
+
+  for (const ruleOption of ruleOptions) {
+    if (typeof ruleOption === 'string') {
+      protectedRules[ruleOption] = '*';
+    }
+
+    if (Array.isArray(ruleOption)) {
+      protectedRules[ruleOption[0]] = ruleOption[1].allowed;
+    }
```

Comment:
What is this all about? Is this future functionality that would allow disabling a rule in specific cases? It's not used now and I don't think we will need it in the future so I'd prefer that we removed it and simplify the rule to just what we need now.

Reply:
The code still has unused functionality for only protecting rules in specific paths that I think we should remove.

Reply:
removed it